### PR TITLE
Fix issue #2807 (get model state from cfg instead of args)

### PR DESCRIPTION
**Pull request by amant555**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes # 2807.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
YUP 🙃

### CODE REVIEW

File: fairseq/models/wav2vec/wav2vec2_asr.py [hunk 1]
```diff
@@ -328,7 +328,7 @@ def __init__(self, args, tgt_dict=None):
             state = checkpoint_utils.load_checkpoint_to_cpu(
                 args.w2v_path, arg_overrides
             )
-            w2v_args = state["args"]
+            w2v_args = state["cfg"]["model"]
```
Comment (alexeib) [inline fairseq/models/wav2vec/wav2vec2_asr.py:331]: can you change this to 
``` w2v_args = state.get("args", None) or state["cfg"]["model"] ```

thanks for the fix!

Comment (amant555) [reply fairseq/models/wav2vec/wav2vec2_asr.py:331]: Yup I did the same ``` state["cfg"]["model"] ```.
Happy to help😁

Comment (alexeib) [reply fairseq/models/wav2vec/wav2vec2_asr.py:331]: great. please push your changes and then ill merge them

Comment (amant555) [reply fairseq/models/wav2vec/wav2vec2_asr.py]: Done!

Comment (alexeib) [reply fairseq/models/wav2vec/wav2vec2_asr.py]: ah sorry i already merged this as part of another fix i was doing earlier today

Comment (amant555) [reply fairseq/models/wav2vec/wav2vec2_asr.py]: No issues.

**Comment by myleott** [comment]
Thanks again for the jumping in and sending the PR! Looks like @alexeib merged this in 65b02d529a45f687da8bbc6ec37611b8a9c96297

**Pull request closed without merging**